provd/internal/services/user/user_test.go

```diff
@@ -34,13 +38,109 @@ func TestEmptyCreateUserRequest(t *testing.T) {
 func TestEmptyValidateUsernameRequest(t *testing.T) {
 	t.Parallel()
 
-	client := newUserClient(t, "", "")
+	client := newUserClient(t)
 
 	userResp, err := client.ValidateUsername(context.Background(), nil)
 	require.Error(t, err, "ValidateUsername should return an error for nil request")
 	require.Empty(t, userResp, "ValidateUsername should return a nil response for a nil request")
 }
 
+func TestDbusObjectsAvalible(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		invalidAccounts bool
+		invalidHostname bool
+		wantErr         bool
+	}{
+		"Invalid accounts object": {
+			invalidAccounts: true,
+			wantErr:         true,
+		},
+		"Invalid hostname object": {
+			invalidHostname: true,
+			wantErr:         true,
+		},
+		"Valid objects": {
+			wantErr: false,
+		},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			t.Cleanup(testutils.StartLocalSystemBus())
+
+			var opts []user.Option
+			if tc.invalidAccounts {
+				opts = append(opts, user.WithAccountsPrefix("invalid"))
+			}
+			if tc.invalidHostname {
+				opts = append(opts, user.WithHostnamePrefix("invalid"))
+			}
+
+			_, err := user.New(testutils.NewDbusConn(t), opts...)
+			if tc.wantErr {
+				require.Error(t, err, "New should return an error, but did not")
+				return
+			}
+			require.NoError(t, err, "New should not return an error, but did")
+		})
+	}
+}
+
+func TestInvalidObjects(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		accountsPrefix string
+		hostnamePrefix string
+		wantErr        bool
+	}{
+		"Invalid accounts prefix": {
+			accountsPrefix: "invalid",
+			wantErr:        true,
+		},
+		"Invalid hostname prefix": {
+			hostnamePrefix: "invalid",
+			wantErr:        true,
+		},
+		"Valid accounts path": {
+			wantErr: false,
+		},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			t.Cleanup(testutils.StartLocalSystemBus())
+			bus := testutils.NewDbusConn(t)
+
+			var opts []user.Option
+			v := reflect.ValueOf(tc)
+			for i := 0; i < v.NumField(); i++ {
+				if v.Field(i).String() != "" {
```

Comment:
This is smart, but not clever :p in the sense that it will be quite hard to maintain. I’m sure you can achieve something very similar, in approximately the same amount of line of codes with what I wrote in the next comment (if none are added meanwhile :p just check for the big comments on test cases)